File: back-end/src/main/java/com/trucks/application/port/ScaleRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import com.trucks.domain.entity.Scale;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public interface ScaleRepository {

    Scale save(Scale scale);

    Optional<Scale> find(UUID id);

    Page<Scale> findAll(PageRequest page);

    Scale update(UUID id, Scale scale);

    List<Scale> deleteAll();
}

File: front-end/src/application/repository/scaleRepository.ts
import { Scale } from "../../domain/entity/scale";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from '../../utils/ull-uuid';

export interface ScaleRepository {
    save: (scale: Scale) => Promise<Either<DataError, Scale>>;

    find: (id: UllUUID) => Promise<Either<DataError, Scale>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Scale[]>>;
    
    deleteAll: () => Promise<Either<DataError, Scale[]>>;
}